### PR TITLE
First try at fixing flang

### LINKED ISSUE #3472: [BUG]: The distributed docs\WindowsLocal.properties has issues with clang
### Describe the bug

CE  gh-2187
I download and installed CE (2187) following the directions in WindowsNative.md.  I installed clang 13.01 at the default directory.
I edited c++.local.properties and changed the MSVC 2017 install version to 14.16.27023,  added the  include and lib paths

compiler.clang_64.includePath=C:\Program Files\LLVM\include
compiler.clang_64.libPath=C:\Program Files\LLVM\lib

However, when I choose clang amd64, the compiler uses the MSVC options 
clang++: error: no such file or directory: '/nologo'
clang++: error: no such file or directory: '/FA'
clang++: error: no such file or directory: '/c'
clang++: error: no such file or directory: '/FaC:\Users\john.mertus\AppData\Local\Temp\compiler-explorer-compiler2022230-2488-137yj45.qt33i\output.s'
clang++: error: no such file or directory: '/FoC:\Users\john.mertus\AppData\Local\Temp\compiler-explorer-compiler2022230-2488-137yj45.qt33i\output.s.obj'
Compiler returned: 1

### Steps to reproduce

The steps are just to install a clean CE on a windows system and follow the steps in doc\WindowsNative.md
Launch and any browser and select clang amd64

### Expected behavior

Compiler returned: 0

### Reproduction link

Not applicable

### Screenshots

Not applicable

### Operating System

Windows 10 v21H2

### Browser version

All

### LINKED ISSUE #3476: [BUG]: Default compilation fails for flang-trunk
### Describe the bug

All Fortran's compilers but `flang-trunk` work as expected. The output of `flang-trunk` is as follows:

```
opt/compiler-explorer/clang-llvmflang-trunk/bin/flang: line 431: gfortran: command not found
flang: in /app, gfortran failed with exit status 127: gfortran -g -S -fparse-only -g -o /app/output.s -S -fparse-only /app/example.f90
ASM generation compiler returned: 127
/opt/compiler-explorer/clang-llvmflang-trunk/bin/flang: line 431: gfortran: command not found
flang: in /app, gfortran failed with exit status 127: gfortran -g -fparse-only -L. -Wl,-rpath,. -g -o /app/output.s -fparse-only -L. -Wl,-rpath,. /app/example.f90
Execution build compiler returned: 127
```

### Steps to reproduce

1. Visit godbolt.org
2. Create Fortran source
3. Choose flang-trunk as a compiler
4. Ensure that compilation fails

### Expected behavior

Compilation succeeds

### Reproduction link

https://godbolt.org/z/zYbsP176M

### Screenshots

image

### Operating System

_No response_

### Browser version

_No response_

### PR DESCRIPTION
**Pull request by RubenRBS**
I currently don't have enough free storage space to test this properly, so I'll do it directly in staging.

Closes #3476

### CODE REVIEW

File: lib/compilers/flang.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+export class FlangCompiler extends FortranCompiler {
+    static override get key() {
+        return 'flang';
+    }
+
+    override getDefaultExecOptions() {
+        const result = super.getDefaultExecOptions();
+        const gfortranPath = this.compilerProps('gfortranPath');
+        if (gfortranPath) {
+            result.env.PATH = result.env.PATH + ':' + gfortranPath;
```
Comment (dkm) [inline lib/compilers/flang.ts:36]: Shouldn't we try to use `join()` from `os.path` method instead of string concat with `:` ?

Comment (partouf) [reply lib/compilers/flang.ts:36]: `path.join()` is to make 1 absolute paths from a root and a relative path. That would not work here.

Comment (dkm) [reply lib/compilers/flang.ts:36]: https://nodejs.org/docs/latest/api/path.html#pathdelimiter  :D

Comment (RubenRBS) [reply lib/compilers/flang.ts:36]: Thanks for the tip, will implement as you suggest :)

Comment (RubenRBS) [reply lib/compilers/flang.ts:36]: We have `:` and `;` in a few places, maybe we could use this somewhere else too!

**Comment by RubenRBS** [comment]
I'll push this to staging this evening and see if it works before merging

**Comment by partouf** [comment]
trying to figure out why this doesn't work, i'm a bit confused

**Comment by partouf** [comment]
Somehow gfortranPath is undefined

**Comment by partouf** [comment]
this is now live

**Pull request merged**